### ISSUE TITLE
RSS feeds do not automatically update

### ISSUE THREAD

**Issue by Fugor** [Novik/ruTorrent#1340]
I'm running 0.9.6 + 0.13.6.

I turned off my RSS filters for a month, then turned those back on, now my feeds do not update.

I can do it manually, and it also triggers my filters perfectly.

I do not see permissions issue or anything else that could explain this.

Thanks.

**Comment by anoma** [comment]
I am seeing this as well however it is as far as I can tell completely random and if you wait long enough it will start working again. I could find no issues in any logs I know off.

**Comment by Fugor** [comment]
I've seen a few other people having this issue too. It's been going for weeks now on my side...
The only explanation I had at first was a permission issue, but I've check all I could on this side and couldn't see anything wrong.
I hope the developers will have a look at this, maybe they could have a hint at what could cause this.

**Comment by Novik** [comment]
I may give to you only one hint, because we have only one difference here. Manual updating of RSS is prepared under web-server user. Automatic - under rtorrent user.

**Comment by anoma** [comment]
Would permissions explain the self healing though. I am absolutely certain that it self heals as I recently went to bed with it broken (after an hour of hunting for the issue) only to return in the AM with a view to further debugging only to find it working.

I can keep an eye out for it happening again if someone can hint where to look for issues.

**Comment by Fugor** [comment]
> I may give to you only one hint, because we have only one difference here. Manual updating of RSS is prepared under web-server user. Automatic - under rtorrent user.

In the wiki it says to use the "web-user" user in the .rctorrent in order to lunch the plugin (among with the others) when rtorrent start. I used to have my rtorrent user. So I changed it according to the wiki to my "web-user", and restarted rtorrent but it didn't solve the issue.

So if I understand you Novik, it means that my rtorrent user is missing some permissions on that plugin ?

Thanks.

**Comment by Novik** [comment]
> In the wiki it says to use the "web-user" user in the .rctorrent in order to lunch the plugin

Launching of plugin - is a different thing. If you open WebUI after rtorrent start, then plugin is launched.
If you place a special line (see wiki) to the rtorrent.rc - then plugin will be launched by automatic way after rtorrent restart, regardless - open you WebUI or not.

> So if I understand you Novik, it means that my rtorrent user is missing some permissions on that plugin ?

For example. Yet once - php process, which do job, may be launched under different users. When you prepare manual update, then this user is a same, as from web-server. For automatic update - this user is a same, as from rtorrent. 
But process is a same, in a both cases.

**Comment by anoma** [comment]
I have managed to catch this is a broken state. Would you like anything debugged/checked before it fixes itself again.

**Comment by Fugor** [comment]
Well on my side:

_ php is run by my www-data.
_ /var/www/rutorrent/share/users/webuser/settings/rss : and everything inside is owned by www-data
_ /var/www/rutorrent/plugins/rss : is owned by www-data
_ ps aux | grep rss : shows rss run by rtorrent user

From your explanations Novik, everything looks ok to me. Am I missing something ?

**Comment by Novik** [comment]
> From your explanations Novik, everything looks ok to me.

Really?
From one side

> _ /var/www/rutorrent/share/users/webuser/settings/rss : and everything inside is owned by **www-data**

From another

> _ ps aux | grep rss : shows rss run by **rtorrent** user

**Comment by Fugor** [comment]
I agree indeed , but to be honest, all my plugins have always been owned by www-data and work fine so far.

Also, even with rtorrent user as the owner and relaunch of rtorrent and refresh rutorrent, it is still the same... Even a 777 -R doesn't help...

**Comment by Fugor** [comment]
Actually:

ps aux | grep rss

Doesn't show it running now...

**Comment by anoma** [comment]
Sorry I am not following. How can unchanging permissions or ownership explain sporadic operation.

Surely with incorrect permissions or ownership it will either always work or never work and not as I have here "sometimes" work and then self heal.

**Comment by Novik** [comment]
Show me
1) Your .rtorrent.rc
2) Output of command
ls -l /var/www/rutorrent/share/users/webuser/settings/rss

**Comment by anoma** [comment]
The container is available here if you wish to look deeper:
https://github.com/linuxserver/docker-rutorrent

However as requested:

https://github.com/linuxserver/docker-rutorrent/blob/a9690d70072570d61f67e02866aaa077204d5707/root/defaults/rtorrent.rc

```
ls -l /usr/share/webapps/rutorrent/plugins/rss
total 124
-rw-rw-r-- 1 abc abc  8097 Oct 29 19:05 action.php
-rw-r--r-- 1 abc abc   237 Oct 29 19:05 conf.php
-rw-rw-r-- 1 abc abc   107 Oct 29 19:05 done.php
-rw-rw-r-- 1 abc abc   429 Oct 29 19:05 ie.css
drwxrwxr-x 1 abc abc    14 Oct 29 19:05 images
-rw-rw-r-- 1 abc abc 50603 Oct 29 19:05 init.js
-rw-rw-r-- 1 abc abc   530 Oct 29 19:05 init.php
drwxrwxr-x 1 abc abc   262 Oct 29 19:05 lang
-rw-rw-r-- 1 abc abc   375 Oct 29 19:05 plugin.info
-rw-rw-r-- 1 abc abc  3905 Oct 29 19:05 rss.css
-rw-r--r-- 1 abc abc 35698 Oct 29 19:05 rss.php
-rw-r--r-- 1 abc abc   330 Oct 29 19:05 update.php
```

**Comment by Novik** [comment]
> ls -l /usr/share/webapps/rutorrent/plugins/rss

Please, read my comment above yet once. I don't need output of your plugin's directory. I need output of your *settings* directory.
For you this should be  /var/www/rutorrent/share/users/[YOUR LOGIN NAME]/settings/rss if you use multiuser configuration and
/var/www/rutorrent/share/settings/rss
otherwise.

**Comment by anoma** [comment]
Neither of those exist as exact matches (the reason i made an incorrect gues previously) so here are all remaining matches:

```
root@e32b724254d3:/$ find -iname rss
./usr/share/webapps/rutorrent/plugins/rss
./config/rutorrent/profiles/settings/rss
./config/rutorrent/profiles/users/abc/settings/rss
```
```
root@e32b724254d3:/$ ls -l /config/rutorrent/profiles/settings/rss
total 0
drwxr-xr-x 2 abc abc 134 Nov  4 09:31 cache

```
```
root@e32b724254d3:/$ ls -l /config/rutorrent/profiles/users/abc/settings/rss
total 0
drwxr-xr-x 2 abc abc 17 Nov  3 13:00 cache
```

**Comment by Fugor** [comment]
```
zordac@ZeBox:~$ ls -l /var/www/rutorrent/share/users/webuser/settings/rss
total 4
drwxrwxrwx 2 rtorrentuser rtorrentuser 4096 Nov  9 13:27 cache
-rwxrwxrwx 1 rtorrentuser rtorrentuser    0 Nov  9 13:11 update.lock
```
rtorrent.rc:

```
scgi_port = 127.0.0.1:5001
encoding_list = UTF-8
port_range = 55000-65000
port_random = yes
check_hash = no
directory = /home/rtorrentuser/downloads
session = /home/rtorrentuser/.session
encryption = allow_incoming, try_outgoing, enable_retry
schedule = watch_directory,1,1,"load_start=/home/rtorrentuser/watch/*.torrent"
schedule = untied_directory,5,5,"stop_untied=/home/rtorrentuser/watch/*.torrent"
use_udp_trackers = yes
dht = off
peer_exchange = no
min_peers = 40
max_peers = 100
min_peers_seed = 10
max_peers_seed = 50
max_uploads = 50
execute = {sh,-c,/usr/bin/php /var/www/rutorrent/php/initplugins.php rtorrentuser &}
schedule = espace_disque_insuffisant,1,30,close_low_diskspace=500M

```

**Comment by Novik** [comment]
```
-rwxrwxrwx 1 rtorrentuser rtorrentuser    0 Nov  9 13:11 update.lock
```

Remove this file.

**Comment by Novik** [comment]
@anoma - You have a strange situation.

This - 
```
root@e32b724254d3:/$ ls -l /config/rutorrent/profiles/settings/rss
total 0
drwxr-xr-x 2 abc abc 134 Nov  4 09:31 cache
```
is a settings directory for the single-user configuration. 

This - 
```
root@e32b724254d3:/$ ls -l /config/rutorrent/profiles/users/abc/settings/rss
total 0
drwxr-xr-x 2 abc abc 17 Nov  3 13:00 cache
```
is a settings directory for the multi-user configuration. I.e. you have both, and date of directory for single user > corresponding date of directory for multi user.
How this may be? Sometimes you use one, sometimes - another? If this true, then this is a reason of rss auto update behavior. Sometimes (correct settings) it works, sometimes (incorrect) - no.

**Comment by anoma** [comment]
I am not sure how to move forward here with this.

Would you consider running the [docker container linked above](https://github.com/linuxserver/docker-rutorrent) and having a look. I have a direct line to that project and can commit to getting any fixes and suggestions in quickly.

**Comment by Fugor** [comment]
> -rwxrwxrwx 1 rtorrentuser rtorrentuser    0 Nov  9 13:11 update.lock
Remove this file.

I did but no luck.

**Comment by Fugor** [comment]
Since this is probably a permission issue. I might have spotted something. I tried to change my rss folder's owner from www-data to rtorrent-user.
But I noticed it always reverse to www-data for some reason (after feeds are manually updated ?). Also my rtorrent-user is part of group www-data.
It is the only folder here that's owned by www-data while the others are owned by rtorrent-user.

Does it speak to you at all ?

**Comment by Novik** [comment]
No. If all profile files are readable/writable for the group, then this is normal situation.

**Comment by Fugor** [comment]
Well when it goes back to owner www-data it ends up like this:

http://imgbox.com/Tk0greD7

Since rtorrent-user is part of www-data, it seems ok to me.

**Comment by Fugor** [comment]
Ok... So I've made a fresh install. I've formatted the whole thing and did a clean install following this guide:

https://jes.sc/kb/rTorrent+ruTorrent-Seedbox-Guide.php

My RSS plugin is empty, I add an RSS link and I get:

[11.12.2016 14:53:07] Erreur lors du chargement du flux.

There's still a permission issue. Everything else works just fine. This is nonsense. By default the permissions are 777 on the rss folder (WTF ?). And owner is www-data.

My user is also part of www-data group. What are the needed permission ? If 777 the whole thing doesn't help, I don't see what could be the problem. 

Thanks.
